azure-mgmt-graph-rbac/src/main/java/com/microsoft/azure/management/graphrbac/implementation/ServicePrincipalImpl.java

```diff
@@ -69,16 +69,17 @@ public ServicePrincipalImpl withAccountEnabled(boolean enabled) {
 
     @Override
     public ServicePrincipal refresh() {
-        return null;
+        setInner(client.list(String.format("servicePrincipalNames/any(c:c eq '%s')", name())).get(0));
+        return this;
     }
 
     @Override
     public Observable<ServicePrincipal> createResourceAsync() {
-        return null;
+        throw new UnsupportedOperationException("Will be implemented in a next release");
```

Comment:
ah, ok. Well, but if a bus runs you over in the meantime (which I hope it doesn't!) there is a chance we'll ship it as is, so...